.github/workflows/release.yaml

```
@@ -0,0 +1,33 @@
+name: Release
+on:
+  release:
+    types: [published]
+
+jobs:
+  release:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    if: github.repository == 'pytest-dev/pytest-timeout'
```

Hi! 👋 

Yep, as said in https://github.com/python-attrs/attrs/pull/1179, could be left out here as it's possible but unlikely forks will trigger `release`. (It's more useful when deploying to Test PyPI for merges to `main`.)

I also normally just check for repo owner, then it's easier to copy/compare between pytest repos.

```suggestion
    if: github.repository_owner == 'pytest-dev'
```